### PR TITLE
JAVA-1765: Update dependencies to latest patch versions

### PR DESCRIPTION
**Pull request by tolbertam**
For [JAVA-1765](https://datastax-oss.atlassian.net/browse/JAVA-1765):

* jnr-ffi, 2.0.7 -> 2.0.9
* jnr-posix, 3.0.27 -> 3.0.44
* metrics-core, 3.2.2 -> 3.2.6
* netty, 4.0.47.Final -> 4.0.56.Final
* netty-tcnative, 2.0.1.Final -> 2.0.7.Final
* HdrHistogram, 2.1.9 -> 2.1.10
* ~snappy-java, 1.1.2.6 -> 1.1.7.1~ (newer versions require java 1.7)
* jackson-core, 2.8.8 -> 2.8.11
* jackson-databind, 2.7.9.2 -> 2.7.9.3
* joda-time, 2.9.1 -> 2.9.9
* lz4-java, 1.3.0 -> 1.4.1 (also new coordinates)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,25 +45,25 @@
...
         <log4j.version>1.2.17</log4j.version>
         <slf4j.version>1.7.25</slf4j.version>
         <slf4j-log4j12.version>1.7.25</slf4j-log4j12.version>
         <guava.version>19.0</guava.version>
-        <netty.version>4.0.47.Final</netty.version>
+        <netty.version>4.0.56.Final</netty.version>
         <metrics.version>3.2.2</metrics.version>
-        <snappy.version>1.1.2.6</snappy.version>
+        <snappy.version>1.1.7.1</snappy.version>
         <lz4.version>1.3.0</lz4.version>
```
Comment (adutra) [inline pom.xml]: LZ4 is now available under new GAV coordinates:

```
<dependency>
    <groupId>org.lz4</groupId>
    <artifactId>lz4-java</artifactId>
    <version>1.4.1</version>
</dependency>
```
The GAV changed but not the classes, and the bytecode level is still Java 6. I was even able to run the OSGi test with LZ4 so I think we could upgrade the dependency.

File: pom.xml [hunk 2]
```diff
@@ -45,25 +45,25 @@
...
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-        <cassandra.version>3.11.1</cassandra.version>
+        <cassandra.version>3.11.2</cassandra.version>
         <java.version>1.6</java.version>
         <log4j.version>1.2.17</log4j.version>
         <slf4j.version>1.7.25</slf4j.version>
         <slf4j-log4j12.version>1.7.25</slf4j-log4j12.version>
         <guava.version>19.0</guava.version>
-        <netty.version>4.0.47.Final</netty.version>
+        <netty.version>4.0.56.Final</netty.version>
```
Comment (adutra) [inline pom.xml:54]: I'm seeing an error with SSL tests on both Mac and Windows:
```
Caused by: java.lang.IllegalArgumentException: Failed to load any of the given libraries: [netty_tcnative_windows_x86_64, netty_tcnative_x86_64, netty_tcnative]
	at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:93)
	... 41 more
	Suppressed: java.lang.UnsatisfiedLinkError: could not load a native library: netty_tcnative_windows_x86_64
		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:205)
		at io.netty.util.internal.NativeLibraryLoader.loadFirstAvailable(NativeLibraryLoader.java:85)
		... 43 more
	Caused by: java.io.FileNotFoundException: META-INF/native/netty_tcnative_windows_x86_64.dll
		at io.netty.util.internal.NativeLibraryLoader.load(NativeLibraryLoader.java:161)
```
Upgrading netty-tcnative to 2.0.7.Final solves the problem.

**Review by adutra — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (tolbertam) [reply pom.xml:54]: D'oh yes, good catch!  I forgot to upgrade tcnative.  Will take care of that and also update the relevant docs as well.

File: pom.xml (hunk 1 above)
Comment (tolbertam) [reply pom.xml]: Good catch, they must not have published the new coordinates to the existing artifacts.  Will update.

**Comment by tolbertam** [comment]
pushed changes based on feedback, will see how appveyor & jenkins goes goes w/r/t platform specific netty-tcnative behavior.

**Review by adutra — APPROVED** [review]

**Comment by adutra** [comment]
Added two more commits with small nits that I spotted. If CI is happy, then ok to merge 👍

**Comment by tolbertam** [comment]
I noticed the OSGi tests failed but hadn't fixed them yet, beat me to it, thanks!

**Comment by tolbertam** [comment]
It appears that snappy 1.1.7+ doesn't currently work with OSGi (xerial/snappy-java#210).  Downgraded to 1.1.4 for now.

**Comment by tolbertam** [comment]
And also just realized snappy 1.1.4+ compiles targeting Java 1.7, so will stick to 1.1.2.6 for now.

**Review by olim7t — APPROVED** [review]
The manual refers to the versions in a couple of places, I updated it.

**Pull request merged**